components/BottomNavigation/tests/snapshot/MDCBottomNavigationBarControllerSnapshotTests.m

```diff
@@ -0,0 +1,222 @@
+    }
+    cellColors = [mutableColors copy];
+  });
+  UICollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"cell" forIndexPath:indexPath];
+  cell.backgroundColor = cellColors[indexPath.row % cellColors.count];
+  return cell;
+}
+
+@end
+
+/** Snapshot tests for @c MDCBottomNavigationBarController. */
+@interface MDCBottomNavigationBarControllerSnapshotTests : MDCSnapshotTestCase
+
+/** The view controller being tested. */
+@property(nonatomic, strong) MDCBottomNavigationBarController *navBarController;
+
+/** A child view controller that has scrollable content. */
+@property(nonatomic, strong) MDCBottomNavigationBarControllerSnapshotTestScrollableChildViewController *scrollableChildVC;
+
+/** A child view controller with non-scrolling content. */
+@property(nonatomic, strong) MDCBottomNavigationBarControllerSnapshotTestFixedContentChildViewController *fixedContentChildVC;
+
+@end
+
+@implementation MDCBottomNavigationBarControllerSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  //  self.recordMode = YES;
+
+  self.navBarController = [[MDCBottomNavigationBarController alloc] init];
+  self.navBarController.navigationBar.barTintColor = [UIColor.redColor colorWithAlphaComponent:(CGFloat)0.5];
+  self.navBarController.view.backgroundColor = UIColor.whiteColor;
+
+  self.fixedContentChildVC = [[MDCBottomNavigationBarControllerSnapshotTestFixedContentChildViewController alloc] init];
+  self.fixedContentChildVC.view.backgroundColor = UIColor.blueColor;
+  self.fixedContentChildVC.tabBarItem.title = @"Fixed";
+
+  UICollectionViewFlowLayout *flowLayout = [[UICollectionViewFlowLayout alloc] init];
+  flowLayout.estimatedItemSize = kSizeOfCellInScrollableChild;
+  self.scrollableChildVC = [[MDCBottomNavigationBarControllerSnapshotTestScrollableChildViewController alloc] initWithCollectionViewLayout:flowLayout];
+  self.scrollableChildVC.collectionView.backgroundColor = UIColor.blueColor;
+  self.scrollableChildVC.tabBarItem.title = @"Scrollable";
+
+  self.navBarController.viewControllers = @[ self.scrollableChildVC, self.fixedContentChildVC ];
+  self.navBarController.navigationBar.unselectedItemTintColor = UIColor.blackColor;
+  self.navBarController.navigationBar.titleVisibility = MDCBottomNavigationBarTitleVisibilityAlways;
+}
+
+- (void)tearDown {
+  self.scrollableChildVC = nil;
+  self.fixedContentChildVC = nil;
+  self.navBarController = nil;
+
+  [super tearDown];
+}
+
+- (void)generateAndVerifySnapshot {
+  UIView *backgroundView = [self.navBarController.view mdc_addToBackgroundView];
+  [self snapshotVerifyView:backgroundView];
+}
+
+#pragma mark - Tests
+
+- (void)testNonScrollingChildViewControllerDefault {
+  // Given
+  self.navBarController.view.bounds = CGRectMake(0, 0, 240, 360);
```

Comment:
This frame is re-created in every test. Please either move it to the set up, or create a static property for it on top

Reply:
Done.